upgrade-docs/0.17.md

```diff
@@ -208,6 +208,32 @@ The type of `main` has changed from `Signal Html` to `Program flags`. The main v
 
 Talking to JavaScript still uses ports. It is pretty similar, but adapted to fit nicely with commands and subscriptions.
 
+### Port modules
+For house keeping sake there's also a new concept called `port module`. All ports must be defined inside a `port module` and the recommendation is to stick to one per project. 
+A `port module` is declared as follows:
+
+```elm
+port module Ports exposing (..)
+```
+
+If you forget to do this, Elm provides you with a friendly compiler message:
+
+```
+-- BAD PORT ------------------------------------------------------- src/Main.elm
+
+You are declaring port `focus` in a normal module.
+
+30| port focus : String -> Cmd msg
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+All ports must be defined in a `port module`. You should probably have just one
+of these for your project. This way all of your foreign interactions stay
+relatively organized.
+
+Detected errors in 1 module.
+```
+
+### Incomming / Outgoing ports
```

Comment:
`Incomming` <- small typo

(disclaimer: I'm not a committer, just an user passing by. +1 to improving upgrade docs anyway :) )